src/PointwiseFunctions/AnalyticSolutions/Poisson/Zero.hpp

```diff
@@ -0,0 +1,75 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <limits>
+
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Elliptic/Protocols.hpp"
+#include "Elliptic/Systems/Poisson/Tags.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+class DataVector;
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace Poisson::Solutions {
+
+/// The trivial solution \f$u=0\f$ of a Poisson equation. Useful as initial
+/// guess.
+template <size_t Dim>
+class Zero : public tt::ConformsTo<elliptic::protocols::AnalyticSolution> {
```

Comment:
Since the implementation is basically the same for all systems, can you do something similar to what we did with `PointwiseFunctions/AnalyticSolutions/Hydro/SmoothFlow.?pp`, where there's a generic implementation and then the systems just specialize what they need to? In this case it seems to be just `supported_tags` that I think is different?

Reply:
Are you suggesting to share this implementation between the elliptic systems? For Poisson and Elasticity this could work I think, since the trivial solution is to set all fields to zero. For XCTS the trivial solution involves some fields are 1 (like the conformal factor) and I would call this `Flat` instead of `Zero`. Do you still think I should combine the classes?

Reply:
That sound similar to what we do with hydro still, yea. Basically, we add extra functions to set the variables that either aren't defined or are defined incorrectly in the base/shared class

Reply:
Hm I'm actually not sure the base class would make the code shorter or easier to understand. The subclasses would need to pull in constructors from the base, forward the variables function and still define equality operators.. Though I agree with you that it would be good to derive from a base class if any non-trivial logic could be shared, I think here it is not worth the effort and would actually make the code longer and more complicated.

Reply:
Sorry, how would the 1 line for constructors and 1 line for `variables` be longer?

```
using shared_elliptic::options;
using shared_elliptic::shared_elliptic;
using shared_elliptic::variables;
using shared_elliptic::pup;
```
that seems like the "long" implementation here

Reply:
Won't I need to implement `variables` and forward, because it's a template, and also define equality operators?

Reply:
I'll also need tests for both derived classes _and_ the base class

Reply:
No, I would template the base class on the `supported_tags` and then you can use the exact implementation you have here, I think. 

Yes, you need tests for both.

If you feel that strongly that it would make the code harder to maintain, you can leave it :)